platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -35,20 +40,47 @@ public class DiagnosticToProblemListener implements DiagnosticListener<JavaFileO
 
     private final ProblemReporter problemReporter;
 
+    /** A collection of filters that will serve as a predicate to determine if a diagnostic should be reported or not.*/
+    private final Collection<Predicate<Diagnostic<? extends JavaFileObject>>> diagnosticFilters = new ArrayList<>();
+
+    /**
+     * A map of explicit overrides between {@link Diagnostic.Kind} and {@link Severity}.
+     * <p>
+     * This will take precedence over the default mapping defined in {@link #mapKindToSeverity(Diagnostic.Kind)}.
+     */
```

Comment:
💭 These javadocs would probably be better on the public methods, than hidden internally on the fields.

Reply:
You have a point, I will put them there

Reply:
Moved!